client/app/components/FavoritesControl.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { $rootScope } from '@/services/ng';
+
+function toggleItem(event, item, callback) {
+  event.preventDefault();
+  event.stopPropagation();
+
+  const action = item.is_favorite ? item.$unfavorite.bind(item) : item.$favorite.bind(item);
+  const savedIsFavorite = item.is_favorite;
+
+  action().then(() => {
+    item.is_favorite = !savedIsFavorite;
+    $rootScope.$broadcast('reloadFavorites');
+    callback();
+  });
+}
+
+export function FavoritesControl({ item, onChange }) {
+  const icon = item.is_favorite ? 'fa-star' : 'fa-star-o';
+  const title = item.is_favorite ? 'Remove from favorites' : 'Add to favorites';
+  return (
+    <a
+      href="javascript:void(0)"
+      title={title}
+      className="btn-favourite"
+      onClick={event => toggleItem(event, item, onChange)}
+    >
+      <i className={'fa ' + icon} aria-hidden="true" />
+    </a>
+  );
+}
+
+FavoritesControl.propTypes = {
+  item: PropTypes.shape({
+    is_favorite: PropTypes.bool.isRequired,
+  }).isRequired,
+  onChange: PropTypes.func,
+  // Force component update when `item` changes.
+  // Remove this when `react2angular` will finally go to hell
```

Comment:
![axe](https://user-images.githubusercontent.com/486954/51998385-3e0f7800-24c1-11e9-8ccd-58977566a332.jpg)

Reply:
Probably all of us. Burn it all!

![](https://media.giphy.com/media/17KRDlheRNADS/giphy.gif)